### ISSUE TITLE
[Feat]: Improve cookie consent in cloud

### ISSUE THREAD

**Issue by cakrit** [netdata/netdata-cloud#728]
### Problem

Our current cookie consent is served by a third app and doesn't include an option to opt-out of analytical cookies.

### Figma

https://www.figma.com/file/ZmnmGMHeKmOGv4EClA28D1/Cookies?node-id=0%3A1&t=MAi7ItQKVbDpOjRk-1

### Description

We should provide our own cookie consent form (no dependency on third parties), that will store consent on the user's browser. This means that consent will have to be provided multiple times if the same user uses multiple devices or browsers and that the information on the consent will need to be provided again, if the user clears their browser cache, start an incognito session, we change the application URL etc.

Note for the following that I'm not specifying the precise text to be used, there are tons of examples on the internet for the appropriate language. 

The consent form must provide two options:
- Accept all cookies
- Select which cookies to accept

If the second option is selected,  users can select to 
- Disable tracking cookies, which is equal to Posthog (and GA if we still send anything there).  
- Disable cookies for the in-app Guides and Knowledge Base, which is equal to Gainsight PX
- Disable error tracking cookies, which is equal to Sentry.
- Also, the option for cookies required for the app to function is shown but users can't deselect it 

If any of these choices is bundled with another choice (e.g. via the same GTM integration), we'll need to show them as such for now. But we have to work on splitting them, to allow finer control (i.e. it's not nice to require tracking cookies in order to show in-app guides and knowledge base, so that implementation in GTM would  need to change). 

I can't think of anything else that we may be including, let me know @novykh if you think of something else.

### Importance

must have

### Value proposition

1. GDPR compliance before launching paid plans

**Comment by hugovalente-pm** [comment]
do we also need to provide an option for users to change their selection?

**Comment by hugovalente-pm** [comment]
suggestions, from examples

### Main notification message

```
This website stores cookies on your computer. These cookies are used to collect information about how you interact with our website and allow us to remember you. We use this information to improve and customize your browsing experience and for analytics and metrics about our visitors on this website and other media. To find out more about the cookies we use, see Netdata's [Privacy Policy](https://www.netdata.cloud/privacy/).
```

We should show the following options:
* Accept all (emphasized button)
* Customize (less emphasized button)
   * Strictly necessary cookies
     ```
     These cookies are essential so that you can have the best experience using Netdata Products. Without these cookies Netdata Products may not work as a result.
     ```
  * Functional cookies -- would include on this category those that Chris mentioned _Disable cookies for the in-app Guides and Knowledge Base, which is equal to Gainsight PX_
     ```
     These cookies allow us to remember choices you make through the use of Netdata Products and allow us to give you better and personalized functionalities.
     ```
  * Analytical and tracking
    ```
    These cookies are used to understand how visitors interact with Netdata Products. They help provide non-identifiable information on user activities through Netdata Products, which is valuable to help us improve our product.
    ```
  * Performance and error tracking
  ```
  These cookies are used to track the performance and errors you may experience using Netdata Products. These are key for us to monitor the status of the product and easily identify issues.
  ```

**Comment by hugovalente-pm** [comment]
an example where even the cookies used are listed
![image](https://user-images.githubusercontent.com/82235632/219365091-94b6b193-c315-428a-a096-4f5bc880c794.png)

**Comment by hugovalente-pm** [comment]
figma suggestion https://www.figma.com/file/ZmnmGMHeKmOGv4EClA28D1/Cookies?node-id=0%3A1&t=MAi7ItQKVbDpOjRk-1

**Comment by hugovalente-pm** [comment]
I would say Hotjar isn't needed since we have posthog to track things on the app but not sure it is being used somewhere else. The other I haven't heard about it before (will check what it is).

Maybe better to loop in @andrewm4894 on this as well

**Comment by andrewm4894** [comment]
I think Hotjar was old heatmap stuff we never really used - should be fine to remove that. Pretty sure no one even has a login anymore. Same with Cubicast, think it was a similar session recording type tool. I've never heard of it or anyone using is, [session recording](https://posthog.com/manual/recordings) in posthog is an easy option if we want it (but we have it disabled for now as is some extra cost involved).

So would be ok by me to disable HotJar and Cubicast - all others on the list above we still use.

**Comment by burbuli8ra** [comment]
Thanks a lot @andrewm4894

**Comment by burbuli8ra** [comment]
@andrewm4894 the implementation has been almost finished!

Please note that for **Google Tag Manager** I have followed the [official docs](https://developers.google.com/tag-platform/devguides/consent#tag-manager) that suggest to initialise the app by setting consent to "denied"!
So when the user has not accepted the analytics cookies the website will load with consent denied, eg:
```
window.gtag("consent", "default", {
  "ad_storage": "denied",
  "analytics_storage": "denied"
})
```
and when the user has already accepted the analytics cookies the same code will load with consent granted, eg:
```
window.gtag("consent", "default", {
  "ad_storage": "granted",
  "analytics_storage": "granted"
})
```
Also, when the user accepts or disables the analytics option from the preferences menu the following code is run to update the consent:
```
window.gtag("consent", "update", {
  ad_storage: "granted/denied",
  analytics_storage: "granted/denied",
})

```

Also, on **PostHog** [docs](https://posthog.com/docs/sdks/js#opt-users-out) the suggested solution is to opt-out/opt-in the users according to their consent.

So, for both **Google Tag Manager** and **PostHog** if I haven't gotten something wrong, manual actions need to be made from your side as well within the suite of each service! 😄 
Let's sync tomorrow so you can check it on `testing` env

**Comment by andrewm4894** [comment]
> manual actions need to be made from your side as well within the suite of each service!

I think in FE for PH its just deciding to call `posthog.opt_out_capturing()` or not?

Can discuss and test a few cases on testing easy enough i'd say.

**Comment by andrewm4894** [comment]
Also @hugovalente-pm we might want to discuss if now is time to fully drop GA since if people see Google - they might just turn off all analytics, but not if they see its just PostHog we use for our analytics. 

@cakrit fyi since i know you have strong opinions here too - in terms of do we even need GTM on app.netdata.cloud (i think last reason was maybe for gainsight in app stuff but not sure - i will double check).

**Comment by andrewm4894** [comment]
looks to be working as expected to me - just tested on testing.netdata.cloud and when i disabled analytics i was not seeing any events go to posthog

**Comment by burbuli8ra** [comment]
Just to give a more clean idea on what the current implementation includes, in case we need to make future adjustments or anything I will include here the implemented approach for each service.

### Google Tag Manager
We load the scripts needed and when the page loads we are adding to `dataLayer` object the `default` `consent` option, according to user preferences.
If the user has not yet given consent or has revoked his consent from preferences, the the value is `denied`.
If the user has previously agreed to the cookie policy, then the value is `granted`.

When the user updates his settings or agrees with our cookie policy for the first time we are updating the `dataLayer` object with `updated` `consent` option according to his preferences.

This service is assigned to **Analytics** cookies.

### PostHog
We load the scripts needed and when the page loads we are running the `posthog.opt_out_capturing()` if the user has denied of has not yet given his consent.
When the user accepts cookie policy for the first time or re-enables the option from settings we run `posthog.opt_in_capturing()` which should re-enable the user actions.
If the user has the actions enabled and decides he wants to disable cookies we run `posthog.opt_out_capturing()`.

This service is assigned to **Analytics** cookies.

### GainsightPX / HubSpot
For both services, we load the scripts needed on initial page load and then if the user has not yet agreed to our cookie policy or has declined his consent previously, then we manually make a copy of the related cookies and we delete them.
If the user decides to re-enable the services, then we restore the deleted cookies from the backup we have kept (with 1 day expiration).
If the user revokes his consent from within user settings, then we also keep a backup and manually remove the related cookies.

GainsightPX service is assigned to **Functional** cookies
HubSpot service is assigned to **Analytics** cookies

### Sentry
This one has no cookies set, so we are just not including user details to our data transmitted to Sentry.

This service is assigned to **Performance** cookies

**Comment by cakrit** [comment]
@andrewm4894 I understand what you're saying about GA potentially causing more people to not accept Analytics cookies as described here, but we see that the people concerned about this don't bother with individual cookies and use tracking blockers instead. Posthog and Gainsight may also be blocked by some of these blockers in the future, if they're not being blocked now. 

I don't want to get rid of GA in app.netdata.cloud because it's good to have different sources to compare, when you question some charts. It also provides much better OOB reports for some things than PH will ever be able to provide. Finally, it has the longest history of any app we use, which is also useful at times. 

I haven't seen people complain its use in app.netdata.cloud, I believe it's expected on public sites.

**Comment by cakrit** [comment]
More generally on GTM I believe we still had a couple of things that couldn't be done on the FE without going through a ton of code (maybe custom events or something), but it's been too long and I really don't recall any more. If we think it's worth investigating again, I'd be happy to take another look and discuss.

**Comment by andrewm4894** [comment]
Yeah I had a look at it on testing today and all seems fine to me.

**Comment by andrewm4894** [comment]
@burbuli8ra @hugovalente-pm - i have made the changes on GTM

added these variables on GTM 

![image](https://user-images.githubusercontent.com/2178292/224317284-ba854f43-9ebb-40e1-ae89-0c46aee36b59.png)

and then enabled GTM Consent Overview such that the tag will now only fire when appropriate consents are in the "granted" state. 

![image](https://user-images.githubusercontent.com/2178292/224317646-1e58ad21-5d1f-420b-a485-11a121274345.png)

I tested with GTM in preview on testing and all looks good to me.

**Comment by burbuli8ra** [comment]
Thanks @andrewm4894 ! So, since you have added functional and performance, I will update my code and include them as well!

**Comment by burbuli8ra** [comment]
Released on [v5.14.0](https://github.com/netdata/cloud-frontend/releases/tag/5.14.0)
